bootstrap/config/config.go

```diff
@@ -637,8 +638,13 @@ func CreateProviderClient(
 
 // loadConfigYamlFromFile attempts to read the specified configuration yaml file
 func (cp *Processor) loadConfigYamlFromFile(yamlFile string) (map[string]any, error) {
+	secretProvider := container.SecretProviderExtFrom(cp.dic.Get)
+	if secretProvider != nil {
+		return nil, fmt.Errorf("failed to get secret provider")
+	}
```

Comment:
You can remove this check. It is always present and code before this would fail if it wasn't